src/core/vector/qgsvectorlayerfeatureiterator.cpp

```diff
@@ -903,7 +903,7 @@ bool QgsVectorLayerFeatureIterator::postProcessFeature( QgsFeature &feature )
 
   if ( result && mDistanceWithinEngine && feature.hasGeometry() )
   {
-    result = mDistanceWithinEngine->distance( feature.geometry().constGet() ) <= mDistanceWithin;
+    result = mDistanceWithinEngine->distanceWithin( feature.geometry().constGet(), mDistanceWithin );
```

Comment:
I admit it's more readable than before, but is that also optimized somehow compared with the previous version?

Reply:
I think the intention is that optimisations will be added in a follow up (potentially by calling a newer GEOS method)

Reply:
Exactly